### ISSUE TITLE
Multiple "owners" are allowed when value should go out os scope

### ISSUE THREAD

**Issue by ayushx35** [rust-lang/rust#156382]
I'm quite new to all of this so terribly sorry if this is just due to some misconfiguration / misunderstanding of the entire functionality.

I tried this code:

```rust
let a = 5;
let b = a; // 'a' is copied to 'b'
println!("{}", a); // This works! 'a' is still valid.
```

I expected to see this happen: 
To show an error as 'a' should be out of scope and should not be valid as value of 'a' was assigned to b

Instead, this happened: 
It compiled successfully and it did not show an error and ran successfully 

`rustc --version --verbose`:
rustc 1.94.1 (e408947bf 2026-03-25)
binary: rustc
commit-hash: e408947bfd200af42db322daf0fadfe7e26d3bd1
commit-date: 2026-03-25
host: aarch64-apple-darwin
release: 1.94.1
LLVM version: 21.1.8

<img width="1470" height="956" alt="Image" src="https://github.com/user-attachments/assets/260f1b61-162e-4957-89bd-2686b22728fc" />

**Comment by theemathas** [comment]
This is working as intended. The type `i32` implements the `Copy` trait. This changes how moving values with this type behaves. Instead of making the old place unusable, the old place is now usable, and you end up with two copies of the same value. See https://doc.rust-lang.org/std/marker/trait.Copy.html
